lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -6766,19 +6824,62 @@ void EmitDestructuredObject(ParseNode *lhs,
 
     if (pnode1 != nullptr)
     {
-        Assert(pnode1->nop == knopList || pnode1->nop == knopObjectPatternMember);
+        Js::RegSlot propIdArrLocation = 0;
+        Js::RegSlot computedPropIdArrLocation = 0;
+        if (hasRest)
+        {
+            uint extraAlloc = 0, auxSize = 0;
+            if (UInt32Math::Mul(staticCount, sizeof(Js::PropertyId), &extraAlloc)
+                || UInt32Math::Add(sizeof(Js::PropertyIdArray), extraAlloc, &auxSize))
+            {
+                ::Math::DefaultOverflowPolicy();
+            }
+            Js::PropertyIdArray *propIds = AnewPlus(byteCodeGenerator->GetAllocator(), extraAlloc, Js::PropertyIdArray, staticCount, 0);
+
+            Assert(pnode1->nop == knopList || pnode1->nop == knopObjectPatternMember || pnode1->nop == knopEllipsis);
+
+            uint32 index = 0;
+            Parser::ForEachItemInList(pnode1, [&](ParseNodePtr current) {
+                if (current->nop != knopEllipsis)
+                {
+                    ParseNodePtr nameNode = current->AsParseNodeBin()->pnode1;
+                    EmitObjectPropertyIdToArray(byteCodeGenerator, propIds->elements, nameNode, &index, staticCount, &hasComputedProps);
+                }
+            });
+
```

Comment:
I feel like this whole thing can move to the EmitObjectPropertyIdToArray function. That way it is easier to understand.